src/Compilers/CSharp/Portable/Syntax/SyntaxKind.cs

```diff
@@ -13,199 +13,385 @@ public enum SyntaxKind : ushort
         List = GreenNode.ListKind,
 
         // punctuation
+        /// <summary>Represents <c>~</c> token.</summary>
         TildeToken = 8193,
+        /// <summary>Represents <c>!</c> token.</summary>
         ExclamationToken = 8194,
+        /// <summary>Represents <c>$</c> token.</summary>
         DollarToken = 8195,
```

Comment:
This will probably be misunderstood. The first thing that will come to mind is interpolation string, but they don't generate this token. What actually this token is generated for? Should I add a note for that?

Reply:
I looked for places where `DollarToken` is used on [sourceroslyn.io](http://sourceroslyn.io/) and found one non-trivial use:

https://github.com/dotnet/roslyn/blob/fd0e8520a8c01977e21772bd825526cb41e88a41/src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs#L161-L165

I think that explains what it is: special token used by the debugger.

Reply:
@svick Thanks!

```suggestion
        /// <summary>Represents <c>$</c> token. It's a debugger special punctuation and not related to string interpolation.</summary>
        DollarToken = 8195,
```